app/Http/Controllers/Admin/Activity/RecipientRegionController.php

```diff
@@ -76,4 +77,31 @@ public function update(RecipientRegionRequest $request, $id): JsonResponse|Redir
             return redirect()->route('admin.activity.show', $id)->with('error', 'Error has occurred while updating recipient-region.');
         }
     }
+
+    /**
+     * append freeze and info_text if recipient country or region exists in any one of the activity transactions
+     * if exists then it freezes the section.
+     *
+     * @param $activity
+     *
+     * @throws JsonException
+     *
+     * @return array
+     */
+    public function getRecipientRegionManipulatedElementSchema($activity): array
```

Comment:
this code can be reused for both recipient region and country. considering adding the test to activity service